renderer/GLSLProgram.h

```diff
@@ -0,0 +1,109 @@
+/*****************************************************************************
+The Dark Mod GPL Source Code
+
+This file is part of the The Dark Mod Source Code, originally based
+on the Doom 3 GPL Source Code as published in 2011.
+
+The Dark Mod Source Code is free software: you can redistribute it
+and/or modify it under the terms of the GNU General Public License as
+published by the Free Software Foundation, either version 3 of the License,
+or (at your option) any later version. For details, see LICENSE.TXT.
+
+Project: The Dark Mod (http://www.thedarkmod.com/)
+
+******************************************************************************/
+
+#ifndef __GLSL_PROGRAM_H__
+#define __GLSL_PROGRAM_H__
+
+class GLSLProgram {
+public:
+	static GLSLProgram * Load( const char *vertexSourceFile, const char *fragmentSourceFile, const char *geometrySourceFile = nullptr );
+	static GLSLProgram * Load( const idDict &defines, const char *vertexSourceFile, const char *fragmentSourceFile, const char *geometrySourceFile = nullptr );
+
+	GLSLProgram( GLuint program );
+	~GLSLProgram();
```

Comment:
Since GLSLProgram owns program object, it makes sense to forbid copying --- just in case someone tries =)

Reply:
Yes, absolutely.